test/integration/multicluster/multicluster-traffic/federated_test.go

```diff
@@ -0,0 +1,151 @@
+package multiclustertraffic
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/testutil"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// TestFederatedService deploys emojivoto to two clusters and has the web-svc
+// in both clusters join a federated service. It creates a vote-bot in the
+// source cluster which sends traffic to the federated service and then checks
+// the logs of the web-svc in both clusters. If it has successfully issued
+// requests, then we'll see log messages indicating that the web-svc can't
+// reach the voting-svc (because it's not running).
+//
+// We verify that the federated service exists and has no endpoints in the
+// source cluster.
+func TestFederatedService(t *testing.T) {
+	if err := TestHelper.SwitchContext(contexts[testutil.TargetContextKey]); err != nil {
+		testutil.AnnotatedFatalf(t,
+			"failed to rebuild helper clientset with new context",
+			"failed to rebuild helper clientset with new context [%s]: %v",
+			contexts[testutil.TargetContextKey], err)
+	}
```

Comment:
Seems like this is not required given we always specify contexts below.

Reply:
it's required to set the context correctly for creating the dataplane namespace.